packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2386,7 +2386,8 @@ function replaySuspendedUnitOfWork(unitOfWork: Fiber): void {
       // TODO: Consider moving this switch statement into that module. Also,
       // could maybe use this as an opportunity to say `use` doesn't work with
       // `defaultProps` :)
-      const Component = unitOfWork.type;
+      const Component = unitOfWork.tag === FunctionComponent ? unitOfWork.type : unitOfWork.type.render;
+      const secondArg = unitOfWork.tag === FunctionComponent ? undefined : unitOfWork.ref;
```

Comment:
It's easier to read and refactor if we split these into separate cases. Makes it easier to remove forwardRef cases in the future. We already have `updateForwardRef` so it makes sense to also fork `replayFunctionComponent` into `replayForwardRef`

Reply:
Thanks, that's done now, `replayForwardRef` and `replayFunctionComponent` are very similar though, is that a problem?

Reply:
Yeah that's fine. The update and mount function are also very similar.

Reply:
Just pass the type as an argument to replayFunctionComponent. I wouldn't use the similarity of updateForwardRef and updateFunctionComponent as an argument because those should probably be unified, too. It's less about code size (Closure will likely inline them anyway) and more to avoid an accidental divergence in the implementations; they're identical except for how `type` is accessed so it makes sense to me for that part to be shared.